spec/02-integration/05-proxy/04-plugins_triggering_spec.lua

```diff
@@ -1002,120 +1006,1372 @@ for _, strategy in helpers.each_strategy() do
     end)
 
     describe("plugin with run_on", function()
-      lazy_setup(function()
-        if proxy_client then
-          proxy_client:close()
-        end
+      describe("(http)", function()
+        local proxy_ssl_client
+
+        lazy_setup(function()
+          if proxy_client then
+            proxy_client:close()
+          end
+
+          helpers.stop_kong()
+
+          db:truncate("routes")
+          db:truncate("services")
+          db:truncate("plugins")
+
+          do
+            -- never used as the plugins short-circuit
+            local mock_service = assert(db.services:insert {
+              name = "mock-service",
+              host = helpers.mock_upstream_host,
+              port = helpers.mock_upstream_port,
+            })
+
+            local receiving_sidecar = assert(db.services:insert {
+              name = "receiving-sidecar",
+              host = helpers.mock_upstream_host,
+              port = 18443,
+              path = "/status/200",
+              protocol = "https",
+            })
+
+            --
+
+            local first_on_first = assert(db.routes:insert {
+              hosts     = { "first-on-first.org" },
+              protocols = { "http" },
+              service   = mock_service,
+            })
+
+            assert(bp.plugins:insert {
```

Comment:
The `assert` around `bp.*:insert` are not necessary. The blueprints throw an error if an `insert` fails.